src/pushsource/_impl/backend/koji_source.py

```diff
@@ -621,6 +622,100 @@ def _push_items_from_vmi_build(self, nvr, meta):
 
         return out
 
+    def _push_items_from_rhcos_build(self, nvr, meta, archives=None):
+
+        LOG.debug("Looking for core os assembler build on %s, %s", nvr, meta)
+
+        build_id = meta["id"]
+        archives = archives or self._get_archives(build_id)
+
+        for archive in archives:
+            if archive.get("filename") == "meta.json":
+                path = self._pathinfo.typedir(meta, archive["btype"])
+                item_src = os.path.join(path, archive["filename"])
+                with open(item_src) as archive_file:
+                    rhcos_meta_data = json.loads(archive_file.read())
+
+        extra = meta.get("extra") or {}
+        image = extra.get("typeinfo").get("image") or {}
+
+        arch = image.get("arch")
+
+        boot_mode = image.get("boot_mode")
+        if boot_mode is not None:
+            boot_mode = BootMode(boot_mode)
+
+        version = (
+            rhcos_meta_data["coreos-assembler.container-config-git"]["branch"].split(
```

Comment:
You mentioned changing this to cope with the fact that this field is optional in the schema, but this is still going to crash if the field is absent or `None`.  Adding some test data like that could confirm it if needed.

If you want to make this tolerant against the missing field, won't it have to be written something more like this?

```python
branch = (rhcos_meta_data.get("coreos-assembler.container-config-git") or {}).get("branch") or ""
version = branch.split("-")[-1]
```

Reply:
my bad I missed it, sending it again.